azure-mgmt-sql/src/main/java/com/azure/management/sql/implementation/SqlDatabaseImpl.java

```diff
@@ -0,0 +1,932 @@
+        super(name, null, innerObject);
+        Objects.requireNonNull(parentSqlElasticPool);
+        Objects.requireNonNull(sqlServerManager);
+        this.sqlServerManager = sqlServerManager;
+
+        this.sqlElasticPools = new SqlElasticPoolsAsExternalChildResourcesImpl(this.sqlServerManager, "SqlElasticPool");
+        this.isPatchUpdate = false;
+        this.importRequestInner = null;
+    }
+
+    @Override
+    public String id() {
+        return this.inner().getId();
+    }
+
+    @Override
+    public String resourceGroupName() {
+        return this.resourceGroupName;
+    }
+
+    @Override
+    public String sqlServerName() {
+        return this.sqlServerName;
+    }
+
+    @Override
+    public String collation() {
+        return this.inner().collation();
+    }
+
+    @Override
+    public OffsetDateTime creationDate() {
+        return this.inner().creationDate();
+    }
+
+    @Override
+    public String currentServiceObjectiveName() {
+        return this.inner().currentServiceObjectiveName();
+    }
+
+    @Override
+    public String databaseId() {
+        return this.inner().databaseId().toString();
+    }
+
+    @Override
+    public OffsetDateTime earliestRestoreDate() {
+        return this.inner().earliestRestoreDate();
+    }
+
+    @Override
+    public DatabaseEdition edition() {
+        return DatabaseEdition.fromString(this.inner().sku().tier());
+    }
+
+    @Override
+    public long maxSizeBytes() {
+        return Long.valueOf(this.inner().maxSizeBytes());
+    }
+
+    @Override
+    public String requestedServiceObjectiveName() {
+        if (this.inner().sku() != null)
+            return this.inner().sku().name();
+        return this.inner().requestedServiceObjectiveName();
+    }
+
+    @Override
+    public DatabaseStatus status() {
+        return this.inner().status();
+    }
+
+    @Override
+    public String elasticPoolId() {
+        return this.inner().elasticPoolId();
+    }
+
+
+    @Override
+    public String elasticPoolName() {
+        return ResourceUtils.nameFromResourceId(this.inner().elasticPoolId());
+    }
+
+    @Override
+    public String defaultSecondaryLocation() {
+        return this.inner().defaultSecondaryLocation();
+    }
+
+    @Override
+    public boolean isDataWarehouse() {
+        return this.edition().toString().equalsIgnoreCase(DatabaseEdition.DATA_WAREHOUSE.toString());
+    }
+
+    @Override
+    public SqlWarehouse asWarehouse() {
+        if (this.isDataWarehouse()) {
+            if (this.parent() != null) {
+                return new SqlWarehouseImpl(this.name(), this.parent(), this.inner(), this.sqlServerManager);
+            } else {
+                return new SqlWarehouseImpl(this.resourceGroupName, this.sqlServerName, this.sqlServerLocation, this.name(), this.inner(), this.sqlServerManager);
+            }
+        }
+
+        return null;
+    }
+
+    @Override
+    public List<RestorePoint> listRestorePoints() {
```

Comment:
Not sure if we plan to keep consistency of the return types with PagedIterable/PagedFlux for all modules.